CommonTools/RecoAlgos/src/PrimaryVertexAssignment.cc

```diff
@@ -121,14 +121,12 @@ std::pair<int, PrimaryVertexAssignment::Quality> PrimaryVertexAssignment::charge
           (!useTime or dtmin / timeReso < maxDtSigForPrimaryAssignment_))
         iVertex = vtxIdMinSignif;
     } else {
-      // consider only distances to first vertex for association of pileup vertices (originally used in PUPPI)
-      if ((vtxIdMinSignif >= 0) && (std::abs(track->eta()) > fEtaMinUseDz_))
-        iVertex =
-            ((firstVertexDz < maxDzForPrimaryAssignment_ and firstVertexDz / dzE < maxDzSigForPrimaryAssignment_ and
-              track->dzError() < maxDzErrorForPrimaryAssignment_) and
-             (!useTime or std::abs(time - vertices.at(0).t()) / timeReso < maxDtSigForPrimaryAssignment_))
-                ? 0
-                : vtxIdMinSignif;
+      // consider only distances to first vertex for association (originally used in PUPPI)
+      if ((vtxIdMinSignif >= 0) && (std::abs(track->eta()) > fEtaMinUseDz_) and
+          ((firstVertexDz < maxDzForPrimaryAssignment_ and firstVertexDz / dzE < maxDzSigForPrimaryAssignment_ and
+            track->dzError() < maxDzErrorForPrimaryAssignment_) and
+           (!useTime or std::abs(time - vertices.at(0).t()) / timeReso < maxDtSigForPrimaryAssignment_)))
```

Comment:
same comment here about `vertices.at(0)` vs. `vertices[0]`.
Also, it seems that boolean operators `&&` and `and` are mixed here, might be cleaner to just use one or the other.